### PR TITLE
Adapt to ec4j revision c09486350b57597f3a6b4f9e0c57199c34612e64

### PR DESCRIPTION
**Pull request by ppalaga**

### CODE REVIEW

**Comment by angelozerr** [comment]
Thanks a lot @ppalaga !

File: org.eclipse.ec4e/src/main/java/org/eclipse/ec4e/internal/EditorConfigPreferenceStore.java [hunk 1]
```diff
@@ -57,68 +58,69 @@ public void applyConfig() {
 				spacesForTabs = null;
 				Integer oldTabWidth = tabWidth;
 				tabWidth = null;
-				Collection<Property> properties = IDEEditorConfigManager.getInstance().queryOptions(file);
-				for (Property property : properties) {
-					if (!property.isValid()) {
-						continue;
+				QueryResult result = IDEEditorConfigManager.getInstance().queryOptions(file);
```
Comment (angelozerr) [inline org.eclipse.ec4e/src/main/java/org/eclipse/ec4e/internal/EditorConfigPreferenceStore.java:61]: Why you have removed loop for properties? Now we must check if property is null, no?

Comment (ppalaga) [reply org.eclipse.ec4e/src/main/java/org/eclipse/ec4e/internal/EditorConfigPreferenceStore.java:61]: > Why you have removed loop for properties?

`map.get()` is more effective if we are interested in the values by name.

> Now we must check if property is null, no?

You are right, the code should do the null checking. Actually, I was hoping to be able to use `QueryResult.getValue(...)` for cases like this one, but it turned out that the current `QueryResult.getValue(...)` does not look at `Property.isValid()` which we'd need here too.

I tend to prefer adding the `Property.isValid()` check to `QueryResult.getValue(...)` and use it here. WDYT? `QueryResult.getValue(...)` already performs the null check.

Comment (angelozerr) [reply org.eclipse.ec4e/src/main/java/org/eclipse/ec4e/internal/EditorConfigPreferenceStore.java:61]: > I tend to prefer adding the Property.isValid() check to QueryResult.getValue(...) and use it here. WDYT? QueryResult.getValue(...) already performs the null check.

For my case, it's perfect, but perhaps we should add a flag in the QueryResult.getValue(...) to get the the invald property, no?

**Pull request merged**